### PR TITLE
feat: introduce pkg for themes

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Refactor to lift Themes out to a standalone module. The aim here is to further improve the boundary between the stateful single page application and rendering layer. Whilst themes _could_ sit alongside components it is better to think of the themes as a visual treatment that can be applied onto components. 

Next up 🦉 Dark Mode(TM) that no one is asking for.

### CODE REVIEW

**Comment by davehakkens** [comment]
> Next up 🦉 Dark Mode(TM) that no one is asking for. 
😅

File: packages/themes/dist/common/button.d.ts
Comment (chrismclarke) [inline packages/themes/dist/common/button.d.ts:1]: question(blocking)
Do we want to be including the dist folder? 

I assume the answer is probably no, and in which case we need to add to the scripting to ensure it's both recompiled when developing locally and compiled as part of build processes

File: packages/themes/tsconfig.json
Comment (chrismclarke) [inline packages/themes/tsconfig.json:1]: nit(non-blocking)
I'm not sure if we really need the full boilerplate - most editors have intellisense for the properties anyway.

**Review by chrismclarke — CHANGES_REQUESTED** [review]
See a couple of comments in-line.

Also as this has potential for accidentally breaking theme/style imports I'll suggest @davehakkens  takes a visual check just to make sure colours, logos etc. appear as expected

File: packages/themes/dist/common/button.d.ts
Comment (thisislawatts) [reply packages/themes/dist/common/button.d.ts:1]: Good catch @chrismclarke, I've updated to ignore `dist` and reworked each of the workflows to build themes.

**Review by davehakkens — APPROVED** [review]
Looks good to me! (only tried PP thought, will test again when deployed to dev sites)

**Comment by thisislawatts** [comment]
@chrismclarke do you think this should be marked as a breaking change, aka a v2? Originally I was only thinking about this as refactoring but maybe its more meaningful than that.

File: package.json [hunk 1]
```diff
@@ -14,18 +14,20 @@
   "private": true,
   "main": "lib/index.js",
   "scripts": {
-    "start": "concurrently --kill-others --names components,platform --prefix-colors blue,magenta \"yarn start:components\" \"yarn start:platform\"",
+    "start": "concurrently --kill-others --names themes,components,platform --prefix-colors cyan,blue,magenta \"yarn start:themes\" \"yarn start:components\" \"yarn start:platform\"",
```
Comment (chrismclarke) [inline package.json:17]: Thanks for updating all of these, appreciate it's a bit of a pain maintaining the long list (on my deep backlog to revisit nx/other options)

**Review by chrismclarke — APPROVED** [review]
Thanks for the updates @thisislawatts , confirm running locally behaves as expected (can make changes to theme and see live reloaded). 

I can see there's a few additional minor tweaks to CI scripts so I'd say good to go once passing

**Review by chrismclarke — APPROVED** [review]

**Pull request merged**